ext/gd/libgd/wbmp.c

```diff
@@ -44,6 +44,9 @@ getmbi (int (*getin) (void *in), void *in)
       i = getin (in);
       if (i < 0)
 	return (-1);
+      long long chk = (long long)(mbi) << 7;
+      if (chk > INT_MAX || (chk | (long long)(i & 0x7f)) > INT_MAX)
```

Comment:
How can the second condition ever be true? `i & 0x7f` is at most `0x7f`, ORing t hat with `chk` will not allow you to reach the limit.
Note that this bug was already fixed upstream by https://github.com/libgd/libgd/commit/a8f1d5cab0cad2bca2ed88a49c3f3de8585ff19b